autosklearn/ensemble_building/builder.py

```diff
@@ -870,6 +889,27 @@ def fit_ensemble(
         metrics = metrics if metrics is not None else self.metrics
         rs = random_state if random_state is not None else self.random_state
 
+        # Validate that kwargs doesn't have duplicates
+        params = {"task_type", "metrics", "random_state", "backend"}
```

Comment:
Could you already construct a dictionary containing all arguments to the `ensemble_class` here so we don't have the keys that should not be in `ensemble_kwargs` duplicated?

Reply:
I'm not entirely sure what you mean, if we take `metrics` as an example, it's passed as an argument as `metrics=metrics` and in `ensemble_kwargs = {"metrics": metrics}`. Rather than silently favour things, I thought it's better to just raise an error.

However if you rather construct the dictionary an do this overwriting, which would you prefer, to favor what's specified in `ensemble_kwargs` or in `metrics` param?

```python
def f(kwargs, metrics):
    # 1. Raise error if overlap
    if "metrics" in kwargs: raise ValueError()
    
    # 2. Favour kwargs
    kwargs = {**{"metrics": metrics}, **kwargs}
    
    # 2. Favour metrics param
    kwargs = {**kwargs, **{"metrics": metrics}}
```

Reply:
So I made it essentially prefer variables passed to `fit_ensemble` over `__init__` and it prefers things in `kwargs` over named parameters.

The first one is just because its more "local" and the second one because `ensemble_kwargs` are the only way for a user to interact with it at all so I figure their passed things should get priority

https://github.com/automl/auto-sklearn/blob/4609c81a75244be9706dd6c8feec27cdcbc5b7bf/autosklearn/ensemble_building/builder.py#L894-L918

Reply:
I don't think that's really what I meant, but it looks good, too. However, does it work? I just tried
```
In [1]: kwargs1 = {'a': 1}

In [2]: kwargs2 = {'a': 2}

In [3]: def func(a):
   ...:     print(a)
   ...: 

In [4]: func(**kwargs1, **kwargs2)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-d86ac439ca51> in <module>
----> 1 func(**kwargs1, **kwargs2)

TypeError: func() got multiple values for keyword argument 'a'
```
and I think this shows that the current code wouldn't do the job. I meant like
```
    params = {
        k: v
        for k, v in [
            ("task_type", task),
            ("metrics", metrics),
            ("random_state", random_state),
        ]
        if v is not None
    }
    # Check for duplicates
    params = params.update(ensemble_kwargs)
    ensemble = ensemble_class(**params)
```

Reply:
I can add an explicit check anywho just to make sure